client/tailscale/tailscale.go

```diff
@@ -90,6 +93,26 @@ func (c *Client) setAuth(r *http.Request) {
 	}
 }
 
+type nodeKey key.NodePublic
+
+func (k nodeKey) modifyRequest(req *http.Request) {
```

Comment:
type nodeKeyAuth? type noiseAuth?

at least put some comments here about why you're setting this basic auth header?